File: app/src/main/java/me/lukeforit/spaceofaday/data/source/cache/ApodDao.java
package me.lukeforit.spaceofaday.data.source.cache;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import java.util.List;

@Dao
public interface ApodDao {
    String TABLE_NAME = "apod";

    @Query("SELECT * FROM " + TABLE_NAME)
    LiveData<List<ApodEntity>> fetchAll();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAll(List<ApodEntity> apods);

    @Query("select * from " + TABLE_NAME + " where id = :id")
    LiveData<ApodEntity> fetchBy(int id);
}

File: app/src/main/java/me/lukeforit/spaceofaday/data/source/cache/ApodDao.kt
package me.lukeforit.spaceofaday.data.source.cache

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

import io.reactivex.Single

internal const val TABLE_NAME = "apod"

@Dao
interface ApodDao {

    @Query("SELECT * FROM $TABLE_NAME")
    fun fetchAll(): Single<List<ApodEntity>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAll(apods: List<ApodEntity>)

    @Query("select * from $TABLE_NAME where id = :id")
    fun fetchBy(id: Int): Single<ApodEntity>

    @Query("select * from $TABLE_NAME where id = :id")
    fun fetchSynchronyouslyBy(id: Int): ApodEntity
}
